Tests/UnitTests/Core/Seeding/HoughTransformTest.cpp

```diff
@@ -0,0 +1,135 @@
+  planeCfg.nBinsY = 1000;
+
+  // instantiate the peak finder
+  Acts::HoughTransformUtils::PeakFinders::IslandsAroundMaxConfig peakFinderCfg;
+  peakFinderCfg.fractionCutoff = 0.7;
+  peakFinderCfg.threshold = 3.;
+  peakFinderCfg.minSpacingBetweenPeaks = {0., 30.};
+
+  // and map the hough plane to parameter ranges.
+  // The first coordinate is tan(theta), the second is z0 in mm
+  Acts::HoughTransformUtils::HoughAxisRanges axisRanges{-3., 3., -2000., 2000.};
+
+  // create the functions parametrising the hough space lines for drift circles.
+  // Note that there are two solutions for each drift circle and angle
+
+  // left solution
+  auto houghParam_fromDC_left = [](double tanTheta,
+                                   const ActsExamples::DriftCircle& DC) {
+    return DC.y() - tanTheta * DC.z() -
+           DC.rDrift() / std::cos(std::atan(tanTheta));
+  };
+  // right solution
+  auto houghParam_fromDC_right = [](double tanTheta,
+                                    const ActsExamples::DriftCircle& DC) {
+    return DC.y() - tanTheta * DC.z() +
+           DC.rDrift() / std::cos(std::atan(tanTheta));
+  };
+
+  // create the function parametrising the drift radius uncertainty
+  auto houghWidth_fromDC = [](double, const ActsExamples::DriftCircle& DC) {
+    return std::min(DC.rDriftError() * 3.,
+                    1.0);  // scale reported errors up to at least 1mm or 3
+                           // times the reported error as drift circle calib not
+                           // fully reliable at this stage
+  };
+
+  // instantiate the hough plane
+  Acts::HoughTransformUtils::HoughPlane<Acts::GeometryIdentifier::Value>
+      houghPlane(planeCfg);
+
+  // also insantiate the peak finder
+  Acts::HoughTransformUtils::PeakFinders::IslandsAroundMax<
+      Acts::GeometryIdentifier::Value>
+      peakFinder(peakFinderCfg);
+
+  // loop over the true hits
+  for (auto& sh : simHits) {
+    houghPlane.reset();
+
+    for (ActsExamples::DriftCircle& dc : driftCircles) {
+      ActsExamples::muonMdtIdentifierFields idf;
+      idf.multilayer = dc.multilayer();
+      idf.stationEta = dc.stationEta();
+      idf.stationPhi = dc.stationPhi();
+      idf.stationName = dc.stationName();
+      idf.tubeLayer = dc.tubeLayer();
+      idf.tube = dc.tube();
+      auto identifier = compressId(idf);
+      auto effectiveLayer = 3 * (dc.multilayer() - 1) + (dc.tubeLayer() - 1);
+
+      houghPlane.fill<ActsExamples::DriftCircle>(
+          dc, axisRanges, houghParam_fromDC_left, houghWidth_fromDC, identifier,
+          effectiveLayer, 1.0);
+      houghPlane.fill<ActsExamples::DriftCircle>(
+          dc, axisRanges, houghParam_fromDC_right, houghWidth_fromDC,
+          identifier, effectiveLayer, 1.0);
+    }
+
+    // now get the peaks
+    auto maxima = peakFinder.findPeaks(houghPlane, axisRanges);
+
+    for (auto& max : maxima) {
+      // check the Hough Transforms results
+      BOOST_CHECK_CLOSE(max.x, sh.first, 5.);
```

Comment:
for tan(theta), perhaps we go finer than 5 as tolerance? 0.1 or so?

Reply:
I think it is the percentage difference in BOOST_CHECK_CLOSE third argument

Reply:
Right that seems to be in percent which is quite surprising to me.